include/device_vector.h

```diff
@@ -0,0 +1,90 @@
+#pragma once
+
+#include <vector>
+
+#include <quda_api.h>
+#include <malloc_quda.h>
+#include <util_quda.h>
+#include <complex_quda.h>
+
```

Comment:
This file should be properly documented with doxygen.  This container looks like it could be useful in general.